src/liballoc/boxed.rs

```diff
@@ -110,15 +111,58 @@ impl<T> Box<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline(always)]
     pub fn new(x: T) -> Box<T> {
-        box x
+        Box::new_in(x, Global)
```

Comment:
We should leave the `box` syntax for the default allocator:
```suggestion
        box x
```

Reply:
Speaking of which, can we change `exchange_malloc` to `box_alloc_global` or something?